packages/extract-text/index.js

```diff
@@ -19,26 +19,28 @@ function extractText (outputFilePattern, fileType) {
 
   const plugin = new ExtractTextPlugin(outputFilePattern)
 
-  return (context, webpackConfig) => {
-    const loaderConfig = getLoaderConfigByType(context, webpackConfig, fileType)
+  return (context, util) => prevConfig => {
+    const loaderConfig = getLoaderConfigByType(context, prevConfig, fileType)
     const nonStyleLoaders = getNonStyleLoaders(loaderConfig, fileType)
 
-    return {
-      module: {
-        loaders: [
-          {
-            test: context.fileType(fileType),
-            exclude: loaderConfig.exclude,
-            loader: plugin.extract({
-              fallbackLoader: 'style-loader',
-              loader: nonStyleLoaders
-            }),
-            loaders: undefined
-          }
-        ]
-      },
-      plugins: [ plugin ]
-    }
+    // Partial application of `addLoader`, `addPlugin` & `removeLoaders`
+    // Bind them to their job, but don't apply them on a config yet
+
+    const _addLoader = util.addLoader({
+      test: context.fileType(fileType),
+      exclude: loaderConfig.exclude,
+      loader: plugin.extract({
+        fallbackLoader: 'style-loader',
+        loader: nonStyleLoaders
+      })
+    })
+    const _addPlugin = util.addPlugin(plugin)
+    const _removeLoaders = removeLoaders(context.fileType(fileType))
+
+    // Now apply them to the config: Remove the existing loaders for that
+    // file type, add the new loader and add the plugin
+
+    return _addPlugin(_addLoader(_removeLoaders(prevConfig)))
```

Comment:
I mentioned `flow` before, it's a lodash function. It's what you need here (though you probably don't need to pull it in from lodash unless you want to). And I'd say it's a must have. It's a key part of function composition which is what this api *should* drive people to.

I prefer [my implementation](https://github.com/substantial/webpack-parts/blob/d17ab43e887b4b7104b8c708bf10c0d07835107c/index.js#L67-L68) of `flow` because it flattens and removes nils allowing you to compose like this:

```
flow(
  option && doThing(),
  option2 && [doThing1(), doThing2],
  doOtherThing(),
)
```